protostar-rust/tests/data/dispatchers/tests/using_dispatchers.cairo

```diff
@@ -32,23 +35,26 @@ fn advanced_types() {
     calldata.append('token');   // name
     calldata.append('TKN');     // symbol
     calldata.append(18);        // decimals
-    calldata.append(0);         // initial supply high
     calldata.append(1111);      // initial supply low
+    calldata.append(0);         // initial supply high
     calldata.append(1234);      // recipient
 
-    let contract_address = deploy_contract('erc20', @calldata).unwrap();
+    let class_hash = declare('ERC20').unwrap();
+    let prepared = PreparedContract { contract_address: 4567, class_hash: class_hash, constructor_calldata: @calldata };
+    let contract_address = deploy(prepared).unwrap();
     let contract_address: ContractAddress = contract_address.try_into().unwrap();
     let dispatcher = IERC20Dispatcher { contract_address };
     let user_address: ContractAddress = 1234.try_into().unwrap();
     let other_user_address: ContractAddress = 9999.try_into().unwrap();
 
     let balance = dispatcher.balance_of(user_address);
-    assert(balance == 1111_u256, 'balance == 0');
+    assert(balance == 1111_u256, 'balance == 1111');
 
-    dispatcher.transfer(other_user_address, 1000_u256);
+    // TODO: Change that when we support mocking addresses, so we can actually call transfer
+    // dispatcher.transfer(other_user_address, 1000_u256);
```

Comment:
Please leave an issue

Reply:
Added it do #1986 description and will link it here.